docs/docs/APIdocs/Optimizers/OptimMethod.md

```diff
@@ -0,0 +1,90 @@
+## OptimMethod ##
+
+OptimMethod is used to update model gradient parameters.We have defined SGD method, Adagrad method, etc.
+Details about those optim methods, you can refer to [Optim-Methods](https://github.com/intel-analytics/BigDL/tree/master/docs/docs/APIdocs/Optimizers/Optim-Methods).
+Now, method construct parameters(e.g."learningRate") and internal training parameters(e.g."epoch") store in optim method instead of state(since version 0.2.0)
+
+Here is mainly to describe how to use those methods when training
+### Set method ###
+**scala**
+```scala
+optimizer.setOptimMethod(method : OptimMethod[T])
+```
+**python**
+```scala
+optimizer = Optimizer(
+    model,
+    training_rdd,
+    criterion,
+    optim_method,
+    end_trigger,
+    batch_size)
+```
+in python, you can set optim method when creating an optimizer
+
+### Save method ###
+```scala
+method.save(path: String, overWrite: Boolean = false)
+```
+`T`: path to save method  
+`overWrite`: whether to overwrite or not
+
+When training, you can use optimizer.setCheckPoint(for scala) or optimizer.set_checkpoint(for python) to save methods at regular intervals.
+
+### Load method ###
+```scala
+val method = OptimMethod.load(path : String)
+```
+`path`: file of optim method path
+
+### Scala example ###
+Here is an example to train LeNet5 model with a loading method.
+```scala
+val trainingRDD = ...
+val valRDD = ...
+val batchSize = 12
+val methodPath = ...
+// Load optim method
+val method = OptimMethod.load(methodPath)
+// Create an optimizer
+val optimizer = Optimizer(
+  model = LeNet5(classNum = 10),
+  sampleRDD = trainingRDD,
+  criterion = ClassNLLCriterion(),
+  batchSize = batchSize
+).setValidation(Trigger.everyEpoch, valRDD, Array(new Top1Accuracy), batchSize)
+  .setEndWhen(Trigger.maxEpoch(15))
+
+optimizer.setOptimMethod(method) // set optim method
+
+optimizer.setCheckpoint(param.checkpoint.get, checkpointTrigger) // set checkpoint to save model and optim method
+
+val trainedModel = optimizer.optimize()
+```
+
+### Python example ###
+Here is an example to train LeNet5 model with a loading method.
```

Comment:
So Python don's support loading an optimMethod snapshot? If Python don't support it.  Maybe we need to add it into description.